core/src/main/kotlin/com/malinskiy/marathon/Marathon.kt

```diff
@@ -174,14 +175,17 @@ class Marathon(
         try {
             scheduler.stopAndWaitForCompletion()
             onFinish(analytics, deviceProvider, attachmentManager)
+        } catch (up: ReportGenerationException) {
+            // We don't want to catch this. This should fail the execution
+            throw up
```

Comment:
Have you considered doing the same for all `Throwable`s?

Reply:
I couldn't test many other exceptions so I'm not sure how often this will happen.
I updated it so that now, we do the same for all throwables, but we need to monitor the results carefully, in case something unexpected breaks